### ISSUE TITLE
Improve CS8603: "Possible null reference return" for String.IsNullOrWhiteSpace check

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#60304]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Improve-CS8603:-Possible-null-reference/1692235)._

---
With SDKProj Properties:
  
```
<LangVersion>preview</LangVersion>
<Nullable>enable</Nullable>
```
Following line 'return CategoryValue;' shows warning: CS8603: "Possible null reference return". Which is incorrect because of the check.
! [SJTPk8NpYL.png] (https://aka.ms/dc/image?name=Be85d2e351c8b47bcb243f558ca8ccb0c637829312953036950_SJTPk8NpYL.png&tid=e85d2e351c8b47bcb243f558ca8ccb0c637829312953036950)

```
public string? CategoryValue { get; set; }
public string Category {
  get
  {
    if (! String.IsNullOrWhiteSpace(CategoryValue))
      return CategoryValue;
  }
}
```

Following line 'return CategoryValue;' does not return warning. Which is correct because of the check.

```
public string? CategoryValue { get; set; }
public string Category {
  get
  {
    if (! String.IsNullOrEmpty(CategoryValue))
      return CategoryValue;
  }
}
```

---
#### Feedback Bot on 3/15/2022, 08:27 PM: 

Thank you for taking the time to provide your suggestion. We will do some preliminary checks to make sure we can proceed further. You will hear from us in about a week on our next steps.

**Comment by RikkiGibson** [comment]
This warning isn't expected to occur in net5.0 and up.

**Comment by jcouv** [comment]
I wasn't able to repro. The issue likely occurred on a target framework where `IsNullOrWhitespace` wasn't yet correctly annotated, as Rikki pointed out. But now it is correctly annotated.
[sharplab](https://sharplab.io/#v2:CYLg1APgxAdgrgGwQQwEYIKYAIMzZgWAChiABAJiwGFiBvYrLUgZiYEYAGAfmuQBcMAcwD2AJwCeANWQI42WlkEY+AbiwBnZWoC+DJq1KdeAkRKz0ijRkr57GFq1YCWAMywAKAITsOAOgCS6gByiAgA8qIA6gAWTgIAygAOyADGGO5U/EJiUjJyAJT5do5MAOzG2RLSshgqxYz1VqTlAEQtdZZWup3d2kA==) (no warning, as expected)

I'll close a "no repro". Thanks
